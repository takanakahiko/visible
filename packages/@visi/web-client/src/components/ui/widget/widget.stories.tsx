import React from 'react';

import { Widget } from '.';

export default {
  title: 'Widget',
  component: Widget,
};

export const Normal = () => (
  <Widget id="test">
    <Widget.Title>Hello</Widget.Title>
    <Widget.Body>this is the content</Widget.Body>
  </Widget>
);
