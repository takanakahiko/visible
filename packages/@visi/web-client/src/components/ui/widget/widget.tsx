import classNames from 'classnames';
import React, { createContext, useContext } from 'react';

const WidgetContext = createContext<{
  labelId?: string;
}>({});

const useLabelId = () => {
  return useContext(WidgetContext).labelId;
};

export interface TitleProps {
  children: React.ReactNode;
}

const Title = (props: TitleProps) => {
  const { children } = props;
  const id = useLabelId();

  return <div id={id}>{children}</div>;
};

export interface BodyProps {
  children: React.ReactNode;
}

const Body = (props: BodyProps) => {
  const { children } = props;

  return <div className={classNames('py-2')}>{children}</div>;
};

export interface WidgetProps {
  id: string;
  children: React.ReactNode;
  className?: string;
}

export const Widget = ({ children, id, className }: WidgetProps) => {
  return (
    <WidgetContext.Provider value={{ labelId: id }}>
      <div role="region" className={className} aria-labelledby={id}>
        {children}
      </div>
    </WidgetContext.Provider>
  );
};

Widget.Title = Title;
Widget.Body = Body;
