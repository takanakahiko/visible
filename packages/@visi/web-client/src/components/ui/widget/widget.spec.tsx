import { render } from '@testing-library/react';
import React from 'react';

import { Normal } from './widget.stories';

describe('Widget', () => {
  it('matches snapshot', () => {
    const { container } = render(<Normal />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
