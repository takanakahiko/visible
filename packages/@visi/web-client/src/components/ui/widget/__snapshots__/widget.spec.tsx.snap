// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget matches snapshot 1`] = `
<div
  aria-labelledby="test"
  role="region"
>
  <div
    id="test"
  >
    Hello
  </div>
  <div
    class="py-2"
  >
    this is the content
  </div>
</div>
`;
